src/binding/petsc4py/src/petsc4py/PETSc/DMStag.pyx

```diff
@@ -38,22 +38,74 @@ class DMStagStencilLocation(object):
 # --------------------------------------------------------------------
 
 cdef class DMStag(DM):
-
+    """A DM object representing a "staggered grid" or a structured cell complex."""
     StencilType       = DMStagStencilType
     StencilLocation   = DMStagStencilLocation
 
-    def create(self, dim, dofs=None, sizes=None, boundary_types=None, stencil_type=None, stencil_width=None, proc_sizes=None, ownership_ranges=None, comm=None, setUp=False):
-        
+    # TODO: can someone please double check descriptions in Parameters?
+    # TODO: do all See Alsos render? they didn't before
+    def create(
+        self,
+        dim: int,
+        dofs: tuple[int, int] | tuple[int, int, int] | tuple[int, int, int, int] | None = None,
+        sizes: tuple[()] | tuple[int] | tuple[int, int] | tuple[int, int, int] | None = None,
```

Comment:
Can this just be `Sequence[int] | None = None`? Even if user passes `tuple[int, int]`, this would be invalid if `dim` was `3`. This is strictly checked in line 107 (see also `asStagDims`: https://github.com/firedrakeproject/petsc_petsc4py_sprint/blob/31ae00a285d07936cb0e0fd929dab4afd9b3f3ec/src/binding/petsc4py/src/petsc4py/PETSc/petscdmstag.pxi#L134). If we did that, we wouldn't have to worry about `tuple[()]` here. I think the same goes `dofs` and `proc_sizes`.

Reply:
Thanks! Still working on that, and all those complex types might end up being `tuple[int, ...]` in the end...